File: 429. N-ary Tree Level Order Traversal/Solution.py
"""
# Definition for a Node.
class Node(object):
    def __init__(self, val, children):
        self.val = val
        self.children = children
"""
class Solution(object):
    def levelOrder(self, root):
        """
        :type root: Node
        :rtype: List[List[int]]
        """
        result, queue = [], [root]
        while len(queue) != 0 and root != None:
            qsize = len(queue)
            res = []
            for i in range(qsize):
                n = queue.pop(0)
                res.append(n.val)
                queue += n.children
            result.append(res)
        return result

File: 429. N-ary Tree Level Order Traversal/Solution.cpp
/*
// Definition for a Node.
class Node {
public:
    int val = NULL;
    vector<Node*> children;

    Node() {}

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/
class Solution {
public:
    vector<vector<int>> levelOrder(Node* root) {
        vector<vector<int>> result;
        queue<Node*> q; q.push(root);
        while (!q.empty() && root) {
            vector<int> res;
            int qsize = q.size();
            for (int i = 0; i < qsize; i++) {
                Node* n = q.front(); q.pop();
                res.push_back(n->val);
                for (auto & ch : n->children) q.push(ch);
            }
            result.push_back(res);
        }
        return result;
    }
};

